### PR TITLE
Simplify enkf_main_copy_ensemble 

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Cleaning some things I noticed while replacing string_list in 400611c9f2d08e61f726874dc8de7e3f2679e504

## Pre review checklist

- [x] Added appropriate labels

### CODE REVIEW

**Comment by sondreso** [comment]
Have not looked at the changes, but to improve the change log, I think this PR needs a better title before merging :sweat_smile:

**Comment by oyvindeide** [comment]
> Have not looked at the changes, but to improve the change log, I think this PR needs a better title before merging 😅

What??!1 😛 Better now?

File: libres/lib/enkf/enkf_main_manage_fs.cpp [hunk 1]
```diff
@@ -128,60 +128,40 @@ void enkf_main_initialize_from_scratch(
...
-    const int ens_size = enkf_main_get_ensemble_size(enkf_main);
+static void enkf_main_copy_ensemble(const ensemble_config_type *ensemble_config,
+                                    enkf_fs_type *source_case_fs,
+                                    int source_report_step,
+                                    enkf_fs_type *target_case_fs,
+                                    const std::vector<bool> &iens_mask,
+                                    const std::vector<std::string> &node_list) {
     state_map_type *target_state_map = enkf_fs_get_state_map(target_case_fs);
 
     {
```
Comment (ManInFez) [inline libres/lib/enkf/enkf_main_manage_fs.cpp:139]: remove unnecessary

File: libres/lib/enkf/enkf_main_manage_fs.cpp [hunk 2]
```diff
@@ -128,60 +128,40 @@ void enkf_main_initialize_from_scratch(
...
-                 src_iens++) {
-                if (src_iens < iens_mask.size() && iens_mask[src_iens]) {
-                    int target_iens = ranking_permutation[src_iens];
+            enkf_config_node_type *config_node =
+                ensemble_config_get_node(ensemble_config, node.c_str());
+
+            int src_iens = 0;
+            for (auto mask : iens_mask) {
+                if (mask) {
                     node_id_type src_id = {.report_step = source_report_step,
```
Comment (ManInFez) [inline libres/lib/enkf/enkf_main_manage_fs.cpp:148]: source_report_step is also 0 I guess

**Review by ManInFez — APPROVED** [review]
just a minor comment

File: libres/lib/enkf/enkf_main_manage_fs.cpp (hunk 2 above)
Comment (oyvindeide) [reply libres/lib/enkf/enkf_main_manage_fs.cpp:148]: One would think so, and it probably is, but it is possible to provide by the user, so would be a breaking change.

**Comment by oyvindeide** [comment]
test this please

**Comment by oyvindeide** [comment]
test this please

**Comment by oyvindeide** [comment]
test this please

**Pull request merged**